service/resource/deployment/deployment.go

```diff
@@ -4,38 +4,75 @@ import (
 	"fmt"
 
 	"github.com/giantswarm/azuretpr"
+	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/azure-operator/service/key"
 )
 
 const (
-	clusterIDParameterName     = "clusterID"
-	clusterSetupDeploymentName = "cluster-setup"
-	clusterSetupTemplate       = "cluster_setup.json"
-	templateBaseURI            = "https://raw.githubusercontent.com/giantswarm/azure-operator/master/service/arm_templates"
-	templateVersion            = "1.0.0.0"
+	mainDeploymentName          = "cluster-main-template"
+	templatesBaseURI            = "https://raw.githubusercontent.com/giantswarm/azure-operator/%s/service/arm_templates/"
```

Comment:
This looks like we are fetching the template at runtime from the Github API. I noticed recently that especially the raw content API is not very reliable. I am against this. We should rather have a templating library like k8scloudconfig. I seem to not understand how this here works. I checked the deployment resource and noted that there is logic in `ProcessCreateState` which actually belongs to `GetCreateState`. The process methods should only take state and act on it via some client implementation. All state creation and manipulation does not belong there. See https://github.com/giantswarm/azure-operator/blob/746539df00e3d47f5842f6f2f48d57dba44978be/service/resource/deployment/resource.go#L178.

Reply:
For hosting linked templates the standard approaches are GitHub or private blob storage (Azure equivalent of S3). I see the reliability problem with using GitHub. We decided to go with it for now so we can iterate on the template deployment.

We could add a template resource to the operator that uploads the ARM templates to blob storage. That's what we'll be doing for the cloudconfig.

Ack, I'll remove the state manipulation from `ProcessCreateState`.

Reply:
I see. I missed how Azure works. So this just is what it is.